### PR TITLE
Implemented wrapping and unwrapping on the DataView Grid

### LINKED ISSUE #7812: Add alignmment options to the grid?
I assume this might add to an entry in the toolbar and also possibly in the Edit menu.

With some character fields an entry can become very wide.  I have also an extreme situation with the files supplied in #7811 

The Excel file has a wide column (very wide - over 2500 characters).  I would like the Excel options of being able to look at it left-justified, (possibly centred) or right-justified.
I like the feature (also in Excel) of double-clicking at the end to make it full width.  But then it is very awkward to try and get it back to a more reasonable width.  I don't know if there is a way to improve on this.

Word wrap is in Excel, and is also in R.  It is under Prepare > Column: Text > Transform > Wrap.  It works well in that it wraps in the grid, but it doesn't unwrap!  
An example is the data in the library in the package statquotes.  I used wrap 25 and later 70.  The grid still keeps the setting of the wrapping from the smallest that was used.  It doesn't unwrap the cell depth, though the text does.
image

### PR DESCRIPTION
**Pull request by N-thony**
Replaces #7922
Fixes #7812
Fixes partially #8443
@rdstern @derekagorhom please test the wrapping and unwrapping here.

### CODE REVIEW

**Review by rdstern — CHANGES_REQUESTED** [review]
@N-thony wow - well almost.  I will test a bit more, but it seems almost to be nailed!  Well done.  
The "almost" is a shape file.  When I originally import it is ok.

But when I try the workdwrap on the geometry variable in the shape file, then it gives a serious error and closes the file and tells me to restart.  Please could you check and see what is possible here.
If you can find where it is going wrong, then I see (at least) 3 options:
a) You may be able to fix it, so wordwrap is also possible on this type of variable.
b) It may be possible to fix under certain circumstances, e.g. the width must be more than a certain value.
c) You don't allow wordwrap on this type of variable.  (That would also be acceptable)

A second problem on the shape file is when I save it into a databook.  Then I open it again.  This is desirable, so I can make it easy for users to draw maps.  Then the data sheet comes wordwrapped - see below:
That's one of the reasons I was keen to have the wordwrap feature.  So that I could unwrap this variable!  This gives me an error as shown below:

image




However, after this error, it did unwrap, which is great.  So, just not doing that operation, when there are no \n, seems to work!
--------------------------------------------------------------------------------------------
That also made me try worwrap on a list variable, and that had a very surprising result - at least surprising (and attractive) for me.

I tried a new data frame of length 1000 and then the calculator divisors of x1 into x2.  Then wordwrap of x2.

This works, but it changes x2 into a character variable.  Now, if you look at my recent issue #8443 you will see that I am unable to make this transformation any other way.  (Making a list variable into a character variable makes it always NA!  So you may have a solution to part of that issue in your code!  I suggest you leave that point for now, till you see the solution for geometry-type variables.  Later we may choose to excluse list variables from those that can be wordwrapped.

**Review by rdstern — CHANGES_REQUESTED** [review]
@N-thony this is just great.  And almost there I think.  I first tried with list variables and that was excellent.  As you say, you have also corrected the bug, so I can get a new data frame when there are list variables.  That's so good!

It now does detect a geometry variable and labels it with a (G) that I really like.  And I am happy that you can't wordwrap it.  

a) Please correct the spelling!  It says varaible and not variable!
b) I get an error when trying to wordwrap a variable in a dataframe that has missing values in the data.  It was fine with this variable first and then I made one row missing and got an error.

image

This is almost ready to merge!

**Review by rdstern — DISMISSED** [review]
@N-thony this is exciting, both in its own right and in the implications for the use of the spreadsheets in R-Instat.  We need to keep testing it, but @lloyddewit it would be great to get this merged, as soon as you are happy with the code.

File: instat/UserControls/DataGrid/ReoGrid/ucrDataViewReoGrid.vb [hunk 1]
```diff
@@ -102,6 +102,29 @@ Public Class ucrDataViewReoGrid
...
+            Exit Sub
+        End If
+
+        If bApplyWrap Then
+            For i As Integer = 0 To grdData.CurrentWorksheet.ColumnCount - 1
+                If i = iColumnIndex Then
+                    grdData.CurrentWorksheet.AutoFitColumnWidth(i)
+                    Exit For
+                End If
+            Next
```
Comment (lloyddewit) [inline instat/UserControls/DataGrid/ReoGrid/ucrDataViewReoGrid.vb:118]: do we need this `For` loop?
```suggestion
            grdData.CurrentWorksheet.AutoFitColumnWidth(iColumnIndex)
```
If we did the above, then would we still need to check `iColumnIndex` for validity?

File: instat/dlgWordwrap.vb [hunk 2]
```diff
@@ -18,5 +18,105 @@ Imports instat.Translations
...
+        ucrReceiverWrapText.SetRCode(clsWrapOrUnwrapFunction, bReset)
+        ucrPnlTextWrap.SetRCode(clsWrapOrUnwrapFunction, bReset)
+    End Sub
+
+    Private Sub TestOkEnabled()
+        If Not ucrReceiverWrapText.IsEmpty Then
+            ucrBase.OKEnabled(True)
+        Else
+            ucrBase.OKEnabled(False)
+        End If
```
Comment (lloyddewit) [inline instat/dlgWordwrap.vb:91]: shorten? (if you prefer to leave as is, then that's also fine)
```suggestion
        ucrBase.OKEnabled(Not ucrReceiverWrapText.IsEmpty)
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@N-thony Loooks very good, just a couple of small questions

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern there were a couple of small changes, please could you retest/approve? thanks

**Review by rdstern — APPROVED** [review]

**Pull request merged**